decidim-results/app/controllers/decidim/results/results_controller.rb

```diff
@@ -12,7 +12,7 @@ class ResultsController < Decidim::Results::ApplicationController
       private
 
       def results
-        @results ||= search.results.page(params[:page]).per(12)
+        @results ||= search.results.order("title -> '#{I18n.locale}' ASC").page(params[:page]).per(12)
```

Comment:
Maybe we should add an index, can you run `Decidim::Results::Result.order("title -> '#{I18n.locale}' ASC").explain` to see how it performs?